.github/workflows/rebuild_c_files.yml

```diff
@@ -0,0 +1,26 @@
+on:
+  schedule:
+    - cron: 0 0 * * *
```

Comment:
Do I understand correctly that this workflow runs every day, regardless of whether anything has changed, and therefore most of the time the work its doing is unnecessary? If so, that seems kinda wasteful. I guess the problem is that there is no event we can hook to find out when there is a new Python version?

What do you think of making this a manual workflow with instructions in the README on how to run it if/when a new Python version is released?

Reply:
I think the important event is when there's a new Cython version, not Python.

What would the problem be with running it daily even if most days there's no diff? It's free and takes a few seconds, so it shouldn't affect our concurrency limit or any such constrained resource. That was my calculus anyway.

If there's a constrained resource we want to preserve, we could either make this run less frequently (like weekly) or not run it on a schedule at all. Regardless, we also get a reactive PR into any PR where there's a did in the c files.

Reply:
> I think the important event is when there's a new Cython version, not Python.

I see. The description says "whenever there's a new python version"

I don't see much value in running it on a schedule.

> we also get a reactive PR into any PR where there's a did in the c files.

I assume that was meant to say "there's a _change_ in the c files."

Reply:
Oops, sorry for my typos!

The value we get from running it on a schedule is that then there's a high probability that any new PRs won't be met with an unrelated surprise change regarding these C files.

Can you share more of what the downside is of running it on a schedule? I'm still not sure I get it. You said it was wasteful to run the check daily if most of the check decides to do nothing. Can you say more about what you think it's wasteful of? To me it's similar to something like dependabot which checks regularly to see if there are updates and prompts you to apply them when there are.

Reply:
I see. I thought it was going through a build process and then throwing away the result if some condition is not met. You're right that the practical effects on time and our github actions allowances are practically very small and insignificant.

But more generally I wonder if running automated processes that constantly consume compute resources with no valuable outcome (most of the time) are bad for the environment—not on an individual basis, but at scale (if everyone does that) it results in unnecessary energy consumption which could have a negative impact on future generations.

To be clear, I think it would be fine if it could be hooked to a meaningful event (if we can get notified of such an event), but polling on a schedule seems mostly wasteful in this context.